subprojects/snapshots/src/main/java/org/gradle/internal/vfs/VirtualFileSystem.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.vfs;
+
+import org.gradle.internal.snapshot.FileSystemSnapshotVisitor;
+
+import java.util.function.Consumer;
+
+public interface VirtualFileSystem {
+
+    void read(String location, FileSystemSnapshotVisitor visitor);
```

Comment:
Yes, a task will read all of its inputs. Though not at once. At best, it will read all of the locations specified by an input property. The input property may consist of filtered (aka file trees) and non-filtered locations. I don't see the advantage of passing multiple locations in here.